File: app/src/main/java/net/squanchy/search/SearchModule.java
package net.squanchy.search;

import net.squanchy.search.engines.SearchEngines;
import net.squanchy.service.firestore.FirebaseAuthService;
import net.squanchy.service.repository.firestore.EventRepository;
import net.squanchy.service.repository.firestore.SpeakerRepository;

import dagger.Module;
import dagger.Provides;

@Module
class SearchModule {

    @Provides
    SearchEngines searchEngines() {
        return new SearchEngines();
    }

    @Provides
    SearchService searchService(
            EventRepository eventRepository,
            SpeakerRepository speakerRepository,
            SearchEngines searchEngines,
            FirebaseAuthService authService
    ) {
        return new SearchService(eventRepository, speakerRepository, searchEngines, authService);
    }
}

File: app/src/main/java/net/squanchy/search/SearchModule.kt
package net.squanchy.search

import dagger.Module
import dagger.Provides
import net.squanchy.search.algolia.AlgoliaSearchEngine
import net.squanchy.service.repository.AuthService
import net.squanchy.service.repository.EventRepository
import net.squanchy.service.repository.SpeakerRepository

@Module
class SearchModule {

    @Provides
    fun searchService(
        eventRepository: EventRepository,
        speakerRepository: SpeakerRepository,
        authService: AuthService,
        algoliaSearchEngine: AlgoliaSearchEngine
    ): SearchService {
        return SearchService(eventRepository, speakerRepository, authService, algoliaSearchEngine)
    }
}
